addons/mail/models/mail_thread.py

```diff
@@ -835,125 +868,76 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         """
         if not isinstance(message, Message):
             raise TypeError('message must be an email.message.Message at this point')
-        MailMessage = self.env['mail.message']
-        Alias, dest_aliases = self.env['mail.alias'], self.env['mail.alias']
         catchall_alias = self.env['ir.config_parameter'].sudo().get_param("mail.catchall.alias")
         bounce_alias = self.env['ir.config_parameter'].sudo().get_param("mail.bounce.alias")
         fallback_model = model
 
         # get email.message.Message variables for future processing
         local_hostname = socket.gethostname()
-        message_id = message.get('Message-Id')
+        message_id = message_dict['message_id']
 
         # compute references to find if message is a reply to an existing thread
-        references = tools.decode_message_header(message, 'References')
-        in_reply_to = tools.decode_message_header(message, 'In-Reply-To').strip()
-        thread_references = references or in_reply_to
-        reply_match, reply_model, reply_thread_id, reply_hostname, reply_private = tools.email_references(thread_references)
+        thread_references = message_dict['references'] or message_dict['in_reply_to']
+        msg_references = [ref for ref in tools.mail_header_msgid_re.findall(thread_references) if 'reply_to' not in ref]
+        mail_messages = self.env['mail.message'].sudo().search([('message_id', 'in', msg_references)], limit=1)
```

Comment:
Just a note as a reminder: No possibility to post on model based on reference without existing message anymore

Reply:
Indeed. We remove support of 6.1 :) .

Reply:
Note that it is explained in the commit message so that it is logged somewhere.

Reply:
It was more a self reminder for me that I'll need to double check this point, I believe you ;)